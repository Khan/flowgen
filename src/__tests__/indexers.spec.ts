import { compiler, beautify } from "..";
import "../test-matchers";

it("should handle indexers", () => {
  const ts = `
type Map = {
  [key: string]: number
}
type ReadOnlyMap = {
  readonly [key: string]: number
}
`;
  {
    const result = compiler.compileDefinitionString(ts, { quiet: true });
    expect(beautify(result)).toMatchSnapshot();
    expect(result).toBeValidFlowTypeDeclarations();
  }

  {
    const result = compiler.compileDefinitionString(ts, {
      quiet: true,
      inexact: false,
    });
    expect(beautify(result)).toMatchSnapshot();
    expect(result).toBeValidFlowTypeDeclarations();
  }
});
