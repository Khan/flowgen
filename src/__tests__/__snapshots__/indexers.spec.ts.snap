// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle indexers 1`] = `
"declare type Map = {
  [key: string]: number,
};
declare type ReadOnlyMap = {
  +[key: string]: number,
};
"
`;

exports[`should handle indexers 2`] = `
"declare type Map = {
  [key: string]: number,
};
declare type ReadOnlyMap = {
  +[key: string]: number,
};
"
`;
